lib/types/object/index.js

```diff
@@ -118,18 +133,44 @@ internals.Object = class extends Any {
                 }
             }
 
-            if (target[rename.from] === undefined) {
-                delete target[rename.to];
+            if (rename.isRegExp) {
+                if (matchedTargetKeys.length === 0) {
+                    delete target[rename.to];
+                }
+                else {
+                    if (!rename.options.multiple &&
+                        matchedTargetKeys.length > 1) {
+
+                        errors.push(this.createError('object.rename.multiple', { from: matchedTargetKeys, to: rename.to }, state, options));
```

Comment:
So in theory, if `abortEarly` is false, you could get two `object.rename.multiple` for a single rename, that seems weird.

Reply:
That would be the desired result when the target object has two keys that both match the rename `from` regex, right? I agree it's weird, but I don't know what the other option would be.

Reply:
Probably not, your current code is, I think, giving a single error containing all the fields that couldn't be renamed. The problem here is it's erroring on the case that was previously for strings, and also on this line that is made for regexps.

Reply:
Oh I see what you mean now. Not "2 errors" as in the 2 fields, but literally hitting two different error blocks in the code for the same rename object. :+1:

I think completely separating the regex/string code paths will solve that.